frontend/src/app/App.test.tsx

```diff
@@ -230,311 +239,382 @@ describe("App", () => {
-          isOwner: true,
-        }),
-      }),
-    })
-    const wrapper = shallow(<App {...props} />)
+  it("hides the top bar if hideTopBar === true", () => {
+    const wrapper = shallow(<App {...getProps()} />)
+    // hideTopBar is true by default
 
-    expect(wrapper.find(MainMenu).prop("hostMenuItems")).toStrictEqual([
-      { type: "separator" },
-    ])
+    expect(wrapper.find("WithTheme(StatusWidget)").exists()).toBe(false)
+    expect(wrapper.find("ToolbarActions").exists()).toBe(false)
   })
 
-  it("shows hostToolbarItems", () => {
-    const props = getProps({
-      hostCommunication: getHostCommunicationProp({
-        sendMessage: jest.fn(),
-        currentState: getHostCommunicationState({
-          queryParams: "",
-          toolbarItems: [
-            {
-              key: "favorite",
-              icon: "star.svg",
-            },
-          ] as IToolbarItem[],
-        }),
-      }),
-    })
-    const wrapper = shallow(<App {...props} />)
+  it("shows the top bar if hideTopBar === false", () => {
+    const wrapper = shallow(<App {...getProps()} />)
     wrapper.setState({ hideTopBar: false })
 
-    expect(
-      wrapper.find(ToolbarActions).prop("hostToolbarItems")
-    ).toStrictEqual([
-      {
-        key: "favorite",
-        icon: "star.svg",
-      },
-    ])
+    expect(wrapper.find("WithTheme(StatusWidget)").exists()).toBe(true)
+    expect(wrapper.find("ToolbarActions").exists()).toBe(true)
   })
 
-  it("closes modals when the modal closure message has been received", () => {
-    const wrapper = shallow(<App {...getProps()} />)
-    const dialog = StreamlitDialog({
-      type: DialogType.ABOUT,
-      sessionInfo: mockSessionInfo(),
-      onClose: () => {},
-    })
-    wrapper.setState({ dialog })
-    expect(wrapper.find(Modal)).toHaveLength(1)
-    const onModalReset = jest.fn()
-    wrapper.setProps(
-      getProps({
-        hostCommunication: getHostCommunicationProp({
-          currentState: getHostCommunicationState({ forcedModalClose: true }),
-          onModalReset,
-        }),
+  describe("handles HostCommunication messaging", () => {
+    let dispatchEvent: (type: string, event: Event) => void
+    let props: Props
+    let wrapper: ShallowWrapper
+    let instance: App
+    let sendMessageFunc: jest.SpyInstance
+
+    beforeEach(() => {
+      dispatchEvent = mockEventListeners()
+      props = getProps()
+      wrapper = shallow(<App {...props} />)
+      instance = wrapper.instance() as App
+
+      // @ts-expect-error - hostCommunicationMgr is private
+      instance.hostCommunicationMgr.setAllowedOriginsResp({
+        allowedOrigins: ["https://devel.streamlit.test"],
+        useExternalAuthToken: false,
       })
-    )
-    expect(wrapper.find(Modal)).toHaveLength(0)
-    expect(onModalReset).toBeCalled()
-  })
 
-  it("changes scriptRunState and fires withHostCommunication callback when scriptStopRequested signal has been received", () => {
```

Comment:
Is this test obsolete?

Reply:
Correct, these are part of that loop pattern we eliminated and this callback firing is tested in HostCommunicationManager instead.